controllers/host/host_controller.go

```diff
@@ -2057,6 +2056,15 @@ func (r *HostReconciler) UpdateConfigStatus(instance *starlingxv1.Host) (err err
 					instance.Status.Reconciled = false
 					// Update strategy required status for strategy monitor
 					r.CloudManager.UpdateConfigVersion()
+					if hostProfile.Spec.Personality == nil &&
+						profile.Personality != nil {
+						hostProfile.Spec.Personality = profile.Personality
+					} else {
+						msg := fmt.Sprintf(
+							"Missing Personality of host in hostprofile %s",
+							hostProfile.Name)
+						return common.NewSystemDependency(msg)
+					}
```

Comment:
Are you referring to checking the "Personality" as nil when bootstrap? It will be checked before in: [BuildAndValidateCompositeProfile](https://github.com/Wind-River/cloud-platform-deployment-manager/pull/324/files#diff-2b5056258b4f6176fbc6079b10a1db34dcabe4920fa3bb9bb39240160447e045R275)